src/package/package.ts

```diff
@@ -194,6 +208,64 @@ export class Package {
     return convertPackage(pkgId, connection, options, project);
   }
 
+  /**
+   * Download the metadata files for a previously published package version, convert them to source format, and put them into a new project folder within the sfdx project.
+   *
+   * @param project
+   * @param options {@link PackageVersionMetadataDownloadOptions}
+   * @param connection
+   * @returns
+   */
+  public static async downloadPackageVersionMetadata(
+    project: SfProject,
+    options: PackageVersionMetadataDownloadOptions,
+    connection: Connection
+  ): Promise<PackageVersionMetadataDownloadResult> {
+    // Validate the destination path is suitable to extract package version metadata (must be new or empty)
+    const destinationFolder = options.destinationFolder || 'force-app/';
+    const destinationPath = path.join(project.getPath(), destinationFolder);
+    if (!fs.existsSync(destinationPath)) {
+      fs.mkdirSync(destinationPath, { recursive: true });
+    }
+    if (!isDirEmpty(destinationPath)) {
+      throw messages.createError('sourcesDownloadDirectoryNotEmpty');
+    }
+
+    // Get the MetadataZip URL from the MetadataPackageVersion record
+    const { allPackageVersionId } = options;
+    const versionInfo: PackagingSObjects.MetadataPackageVersion = (await connection.tooling
+      .sobject('MetadataPackageVersion')
+      .retrieve(allPackageVersionId)) as PackagingSObjects.MetadataPackageVersion;
+
+    if (!versionInfo.MetadataZip) {
+      throw messages.createError('unableToAccessMetadataZip');
+    }
+
+    const responseBase64 = await connection.tooling.request<string>(versionInfo.MetadataZip, {
+      encoding: 'base64',
+    });
+    const buffer = Buffer.from(responseBase64, 'base64');
+
+    // Unzip the metadata zip file into a tmp directory. It still needs to be converted to source format.
```

Comment:
We should be able to take the response, build a ComponentSet from it in memory, convert it and then write to disk like the SDR library does.  Writing to disk twice should not be necessary.